app/services/the_tvdb_service.rb

```diff
@@ -84,24 +123,16 @@ def build_episode_path(series_id, season_number)
     path
   end
 
-  def season_filter(episodes, season_id)
-    return episodes.select { |ep| ep['airedSeason'].present? } if season_id.blank?
-
-    episodes.select { |ep| ep['airedSeasonID'] == season_id.to_i && ep['airedSeason'].present? }
-  end
-
-  def find_season_number(episodes)
-    episodes.count.zero? ? 1 : episodes.first['airedSeason']
+  def process_episode_data(response, media_id, tvdb_series_id)
+    response['data'].each do |tvdb_episode|
+      row = Row.new(media_id, tvdb_episode, tvdb_series_id)
+      row.update_episode!
+    end
   end
 
-  def create_mapping(media_id, series_id, season_number)
-    mapping = Mapping.where(
-      external_site: 'thetvdb',
-      item_id: media_id,
-      item_type: 'Anime'
-    ).first_or_initialize
-
-    mapping.external_id = "#{series_id}/#{season_number}"
-    mapping.save!
+  def update_anime_episode_count(media_id, tvdb_episodes_count)
+    anime = Anime.find(media_id)
+    anime.update_unit_count_guess(tvdb_episodes_count)
```

Comment:
Is there a good reason for this to be considered a "guessed" episode count?

Reply:
@matthewdias and I had a conversation about this. Being as I wrote that like 2 weeks ago and left myself no notes I honestly don't remember...

Reply:
Iirc the reason was tvdb episode counts just match up with how many episode pages have been added to the DB. For example, if a show is gonna be 26 eps, but info has only been submitted for 6, we'd get 6 as a count. So if ours are nil and we imported those as actual counts we'd have a lot of wrong counts. (At least last I checked this is the case. Might wanna verify that)

Reply:
that sounds right from what I remember.